### PR TITLE
Fix for reading constants, incorrect index used.

### PR DESCRIPTION
**Pull request by galapaegos**
While looking over issues with amplitude 4 body decays, it looks like these indices are incorrect.  They should be referring to 0-based indexing, not 1-based.

Currently do not have a machine to fully test that this resolves issues, waiting for access.

### CODE REVIEW

**Comment by henryiii** [comment]
I believe that the array looks like this:

| N | Radius | M | M1 | M2 | M3 | M4 |     |     |     |     |     |     |
|---|---|---|---|---|---|---|---|---|---|---|---|---|
| 12 | 1.5 | 1.8645 | 0.13957 | 0.13957 | 0.493677 | 0.13957 | 0 | 2 | 6 | 6 | 6 | 36 |

However, I'm not sure where these values are being put in. This changed a bit in 2.2, since before M was not a part of this but was hardcoded (I believe).

**Comment by galapaegos** [comment]
This is correct, that was my mistake.

I did find one issue, it looks like not all constants are being pass in for the line shapes.  This fix is required because the mPair value is read from the LineShape, it is not part of the device_Tddp function.  You can see this in LSCalculator_TD.

The AmpCalc_TD cache is not reading the correct value.  It looks like either the AmpCalc has an incorrect index (unlikely), or the SpinFactor is calculated incorrectly (likely).

File: src/PDFs/physics/Amp4Body_TD.cu [hunk 1]
```diff
@@ -401,16 +406,16 @@ __host__ Amp4Body_TD::Amp4Body_TD(std::string n,
...
+    cachedWidths = new fptype[LineShapes.size()];
 
     for(int i = 0; i < LineShapes.size(); i++) {
         lscalculators.push_back(new LSCalculator_TD());
     }
 
-    // for (int i = 0; i < amp_idx.size(); i++)
-    //    printf("%i - %i\n", i, amp_idx[i]);
+    for(int i = 0; i < amp_idx.size(); i++)
+        printf("%i - %i\n", i, amp_idx[i]);
```
Comment (henryiii) [inline src/PDFs/physics/Amp4Body_TD.cu]: These printouts should be protected by an if debug line.

Comment (galapaegos) [reply src/PDFs/physics/Amp4Body_TD.cu]: Thanks for finding those, those were not supposed to be committed.

**Pull request merged**